scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/kms/common/KeyLoader.scala

```diff
@@ -45,27 +45,51 @@ case class KeyLoader(val fromKms: Boolean,
     protected val CRYPTO_MODE = AES_CBC_PKCS5PADDING
     protected var encryptedDataKey: String = ""
     protected val hadoopConfig = if (config != null) config else new Configuration()
+    protected val enableNativeAESCBC = SparkSession.builder().getOrCreate()
+      .sparkContext.hadoopConfiguration
+      .get("spark.bigdl.enableNativeAESCBC", "false") match {
+        case "true" => true
+        case "false" => false
+    }
+
     // retrieve the plaintext string of an existing data key
     def retrieveDataKeyPlainText(fileDirPath: String): String = {
-        val encryptedDataKey = {
+        val encryptedDataKey = if (enableNativeAESCBC) {
+          // native AES CBC uses file granularity instead of folder
+          val filePath = new Path(fileDirPath).toString
+          val fs: FileSystem = FileSystem.get(new URI(filePath), hadoopConfig)
+          val inStream = fs.open(new Path(filePath))
+          val iv = new Array[Byte](16)
+          inStream.read(iv)
+          new String(iv, StandardCharsets.UTF_8)
+        } else {
           val metaPath = new Path(fileDirPath + "/" + META_FILE_NAME).toString
           val fs: FileSystem = FileSystem.get(new URI(metaPath), hadoopConfig)
           val inStream = fs.open(new Path(metaPath))
           val jsonStr = scala.io.Source.fromInputStream(inStream)
             .takeWhile(_ != null).mkString
           KmsMetaFormatSerializer(jsonStr).encryptedDataKey
         }
+
         val dataKeyPlainText = if (fromKms) {
           kms.retrieveDataKeyPlainText(primaryKeyMaterial, "", null, encryptedDataKey)
         } else {
-          val cipher = dataKeyCipher(Cipher.DECRYPT_MODE)
-          val encryptedDataKeyBytes = Base64.getDecoder().decode(encryptedDataKey)
-          val dataKeyPlainTextBytes = cipher.doFinal(encryptedDataKeyBytes)
-          new String(dataKeyPlainTextBytes, StandardCharsets.UTF_8)
+          if (enableNativeAESCBC) {
```

Comment:
this if can be merged to  the outer one?

Reply:
Not feasible since the three condition branches return three different variables